simd/src/Kokkos_SIMD_Scalar.hpp

```diff
@@ -149,53 +124,76 @@ class simd<T, simd_abi::scalar> {
   KOKKOS_FORCEINLINE_FUNCTION value_type operator[](std::size_t) const {
     return m_value;
   }
+  KOKKOS_FORCEINLINE_FUNCTION simd operator-() const noexcept {
+    return simd(-m_value);
+  }
 
-  KOKKOS_FORCEINLINE_FUNCTION friend constexpr simd operator>>(
+  [[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION friend constexpr simd operator*(
+      simd const& lhs, simd const& rhs) noexcept {
+    return simd(lhs.m_value * rhs.m_value);
+  }
+  [[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION friend constexpr simd operator/(
+      simd const& lhs, simd const& rhs) noexcept {
+    return simd(lhs.m_value / rhs.m_value);
+  }
+  [[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION friend constexpr simd operator+(
+      simd const& lhs, simd const& rhs) noexcept {
+    return simd(lhs.m_value + rhs.m_value);
+  }
+  [[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION friend constexpr simd operator-(
+      simd const& lhs, simd const& rhs) noexcept {
+    return simd(lhs.m_value - rhs.m_value);
+  }
+  [[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION friend constexpr simd operator>>(
       simd const& lhs, int rhs) noexcept {
     return simd(lhs.m_value >> rhs);
   }
-  KOKKOS_FORCEINLINE_FUNCTION friend constexpr simd operator>>(
+  [[nodiscard]] KOKKOS_FORCEINLINE_FUNCTION friend constexpr simd operator>>(
       simd const& lhs, simd const& rhs) noexcept {
-    return simd(lhs.m_value >> static_cast<int>(rhs[0]));
+    return simd(lhs.m_value >> static_cast<value_type>(rhs));
```

Comment:
Why are we casting to `value_type` rather than using `rhs.value` directly?

Reply:
Replaced to use `rhs.m_value` instead. [787c301](https://github.com/kokkos/kokkos/pull/6320/commits/787c3010702bb0005ffbc25091b126a2e23cfbc3)